### DISCUSSION TITLE
Extract the JSX parser from its component into a separate utility and export

### DISCUSSION THREAD

**Discussion by mallison** [TroyAlford/react-jsx-parser#186] [Ideas]
Hello

I've written a custom React reconciler that doesn't target the DOM and I need to be able to parse JSX at runtime. I couldn't find any standalone library to do that until I found this 

For my use case it'd be great to have the JSX parser as a separate class independent of the component so I can generate and element tree and pass to my custom renderer.  I've experimented with this and it works perfectly!

If you think this is a good idea I can submit a PR with the changes.

Thanks for this super useful code!

**Comment by TroyAlford** [comment]
I'd love to take a look :) that sounds like a great abstraction.

**Reply by mallison** [reply]
Created a PR (WIP) https://github.com/TroyAlford/react-jsx-parser/pull/187.

It's possibly a bit of a niche use case.
